arm-billing/2017-02-27-preview/swagger/billing.json

```diff
@@ -0,0 +1,308 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2017-02-27-preview",
+    "title": "BillingClient",
+    "description": "Billing client provides access to billing resources for Azure Web-Direct subscrions. Other subscription types which were not purchased directly through the Azure web portal are not currently supported through this preview API."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Billing/invoices": {
+      "get": {
+        "tags": [
+          "Invoices"
+        ],
+        "operationId": "Invoices_List",
+        "description": "Lists the available invoices for a subscritpion in reverse chronological order beginning with the most recent one.",
+        "externalDocs": {
+          "url": "https://go.microsoft.com/fwlink/?linkid=842057"
+        },
+        "x-ms-examples": {
+					"InvoicesList": {
+    				"$ref": "../examples/InvoicesList.json" 
+        	},
+					"InvoicesExpand": {
+    				"$ref": "../examples/InvoicesExpand.json" 
+        	}
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "$filter may be used to filter invoices by invoicePeriodEndDate. The filter supports 'eq', 'lt', 'gt', 'le', 'ge', and 'and'. It does not currently support 'ne', 'or', or 'not'"
+          },
+          {
+            "name": "$top",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "description": "$top may be used to limit the number of results to the most recent N invoices."
```

Comment:
Consider adding a `minimum` (and `maximum`?) if you want client side validation. I imagine it would be nice to have the client yell at you with a *meaningful* exception for passing negative values and such.